### PR TITLE
Add getrange command

### PR DESCRIPTION
**Pull request by icemint0828**
related to #162 

* add tests.
* add also server command.

Please review when you have time.

### CODE REVIEW

File: cmd/command.go [hunk 1]
```diff
@@ -250,6 +250,24 @@ func mGet(cli *Client, args [][]byte) (interface{}, error) {
 	return values, err
 }
 
+func getRange(cli *Client, args [][]byte) (interface{}, error) {
+	if len(args) != 3 {
+		return nil, newWrongNumOfArgsError("getrange")
+	}
+	key, start, end := args[0], args[1], args[2]
+	s, err := strconv.Atoi(string(start))
```
Comment (roseduan) [inline cmd/command.go:258]: Variable names should be more clear, such as `start`、`end`， otherwise others maybe do not know what it means.

Comment (roseduan) [reply cmd/command.go:258]: start, err := strconv.Atoi(string(arg[1]))

File: strs.go [hunk 2]
```diff
@@ -57,6 +57,41 @@ func (db *RoseDB) MGet(keys [][]byte) ([][]byte, error) {
 	return values, nil
 }
 
+// GetRange returns the substring of the string value stored at key,
+// determined by the offsets start and end.
+func (db *RoseDB) GetRange(key []byte, start, end int) ([]byte, error) {
+	db.strIndex.mu.RLock()
+	defer db.strIndex.mu.RUnlock()
+
+	val, err := db.getVal(db.strIndex.idxTree, key, String)
```
Comment (roseduan) [inline strs.go:66]: if val is nil or `len(val) == 0`, return directly

File: strs.go [hunk 3]
```diff
@@ -57,6 +57,41 @@ func (db *RoseDB) MGet(keys [][]byte) ([][]byte, error) {
...
+	}
+
+	// handles out of range requests by limiting the resulting range to the actual length of the string.
+	if start > len(val)-1 {
+		return []byte{}, nil
+	}
+	if end > len(val)-1 {
+		end = len(val) - 1
+	}
+	return val[start : end+1], nil
```
Comment (roseduan) [inline strs.go:92]: If start and end both are positive numbers, and start > end. eg:
val is "abcde", start = 3, end = 1

val[start:end] will panic

File: cmd/command.go (hunk 1 above)
Comment (icemint0828) [reply cmd/command.go:258]: I understand.
I will fix it.

File: strs.go (hunk 2 above)
Comment (icemint0828) [reply strs.go:66]: Yeah, that's right.

File: strs.go (hunk 3 above)
Comment (icemint0828) [reply strs.go:92]: Oh, I did it.
I had completely misplaced that case.
Thanks for pointing that out!

**Comment by icemint0828** [comment]
Thank you for the review.
I have corrected the points you pointed out and hope you will check it out when you have time.

**Comment by roseduan** [comment]
LGTM

**Pull request merged**